handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -57,7 +60,22 @@
         // If in the classpath, try to load the native library and initialize netty-tcnative.
         if (cause == null) {
             try {
-                NativeLibraryLoader.load("netty-tcnative", SSL.class.getClassLoader());
+                String os = normalizeOs(SystemPropertyUtil.get("os.name", ""));
+                String arch = normalizeArch(SystemPropertyUtil.get("os.arch", ""));
+                String libName = "netty-tcnative-" + os + '-' + arch;
+                try {
+                    // First, try loading the platform-specific library. Platform-specific
+                    // libraries will be available if using a tcnative uber jar.
+                    NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
```

Comment:
@nmittler if I created a pull request with the changes above, would you be for, or against? If against I could potentially work with an uber jar of dynamically-linked openssl libs..

```code
ln -s /usr/lib64/libssl.so.10 /usr/lib64/libssl.so.1.0.0
ln -s /lib64/libcrypto.so.10 /lib64/libcrypto.so.1.0.0
```